pkg/geo/encode_test.go

```diff
@@ -320,3 +320,101 @@ func TestSpatialObjectToGeoHash(t *testing.T) {
 		}
 	})
 }
+
+func TestLineFromEncodedPolyline(t *testing.T) {
+	testCases := []struct {
+		desc         string
+		a            string
+		p            int
+		expectedLine string
+	}{
+		{
+			"EncodedPolyline of three points, including negative values, precision 5",
+			"_p~iF~ps|U_ulLnnqC_mqNvxq`@",
+			5,
+			"SRID=4326;LINESTRING(-120.2 38.5, -120.95 40.7, -126.453 43.252)",
+		},
+		{
+			"EncodedPolyline of three points, including negative values, precision 6",
+			"_p~iF~ps|U_ulLnnqC_mqNvxq`@",
+			6,
+			"SRID=4326;LINESTRING(-12.02 3.85,-12.095 4.07,-12.6453 4.3252)",
+		},
+		{
+			"EncodedPolyline of two points, including negative values, precision 5",
+			"_p~iF~ps|U_ulLnnqC",
+			5,
+			"SRID=4326;LINESTRING(-120.2 38.5,-120.95 40.7)",
+		},
+		{
+			"EncodedPolyline of two points, precision 3",
+			"??_ibE_ibE",
+			3,
+			"SRID=4326;LINESTRING(0 0, 100 100)",
+		},
+		{
+			"EncodedPolyline of four points, including negative values, precision 4",
+			"wkcn@f}gDgfv}@gqcrB~lor@f_ssA}dxgMwujpN",
+			4,
+			"SRID=4326;LINESTRING(-8.65 77.23, 180.0 180.0, 41.35 95.6, 856.2344 843.9999)",
+		},
+		{
+			"EncodedPolyline of three points, precision 1",
+			"_ibE?_ibE?~reK?",
+			1,
+			"SRID=4326;LINESTRING(0 10000, 0 20000, 0 0)",
+		},
+		{
+			"EncodedPolyline of two points, small decimal values, precision 8",
+			"vybw}D_osrst@o~dw}D~nsrst@",
+			8,
+			"SRID=4326;LINESTRING(9 -1.000099,0 0.000011)",
+		},
+		{
+			"EncodedPolyline of two points, precision 5",
+			"ud}|Hi_juBa~kk@m}t_@",
+			5,
+			"SRID=4326;LINESTRING(19.38949 52.09179, 24.74476 59.36716)",
+		},
+		{
+			"EncodedPolyline of two points, malformed point, precision 1",
+			"_ibE_ibE?",
+			1,
+			"SRID=4326;LINESTRING(10000 10000,10000 10000)",
+		},
+		{
+			"EncodedPolyline of three points, malformed point, precision 1",
+			"_ibE?_ibE??",
+			1,
+			"SRID=4326;LINESTRING(0 10000, 0 20000, 0 20000)",
+		},
+		{
+			"EncodedPolyline of three points, precision 1",
+			"_ibE?_ibE???",
+			1,
+			"SRID=4326;LINESTRING(0 10000, 0 20000, 0 20000)",
+		},
+		{
+			"EncodedPolyline of two points, integer values, precision 0",
+			"CACC",
+			0,
+			"SRID=4326;LINESTRING(1 2, 3 4)",
+		},
```

Comment:
for completelyness, can you add an empty string?

Reply:
Good catch, added